src/ray/core_worker/core_worker.h

```diff
@@ -427,6 +427,13 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
                                 rpc::GetCoreWorkerStatsReply *reply,
                                 rpc::SendReplyCallback send_reply_callback) override;
 
+  /// Handle a worker failure notification from GCS pubsub.
+  ///
+  /// \param worker_id The ID of the failed worker.
+  /// \param worker_data Data associated with the worker failure.
+  void HandleWorkerFailed(const WorkerID &worker_id,
+                          const gcs::WorkerFailureData &worker_failed_data);
```

Comment:
We should probably this "HandleWorkerFailedUnexpectedly" or something to clarify the semantics.